cypress/e2e/pages-links.spec.js

```diff
@@ -598,3 +598,113 @@ describe('Page link handling', function() {
 		}
 	})
 })
+
+// Previews got added with Nextcloud 29
+if (!['stable27', 'stable28'].includes(Cypress.env('ncVersion'))) {
+	let previewLinkTargetPageId, pageUrls
+
+	describe('Page link preview handling', function() {
+		before(function() {
+			cy.loginAs('bob')
+			cy.deleteAndSeedCollective('Link Preview Testing')
+				.seedPage('Link Source', '', 'Readme.md')
+				.seedPage('Link Target', '', 'Readme.md').then(({ pageId }) => {
+					previewLinkTargetPageId = pageId
+					pageUrls = [
+						`${baseUrl}/index.php/apps/collectives/Link%20Preview%20Testing/Link%20Target?fileId=${previewLinkTargetPageId}`,
+						`${baseUrl}/index.php/apps/collectives/Link%20Preview%20Testing/Link%20Target`,
+						`${baseUrl}/index.php/apps/collectives/p/qqqYoCgYRnZ598p/Link%20Preview%20Testing/Link%20Target?fileId=${previewLinkTargetPageId}`,
+						`${baseUrl}/index.php/apps/collectives/p/qqqYoCgYRnZ598p/Link%20Preview%20Testing/Link%20Target`,
+					]
+				}).then(() => {
```

Comment:
I think you can avoid the `let` binding - you don't seem to be using it elsewhere and the `pageId` and then the `pageUrls` can be handed from one then block to the next:
```suggestion

	describe('Page link preview handling', function() {
		before(function() {
			cy.loginAs('bob')
			cy.deleteAndSeedCollective('Link Preview Testing')
				.seedPage('Link Source', '', 'Readme.md')
				.seedPage('Link Target', '', 'Readme.md')
				.then(({ pageId }) => [
					`${baseUrl}/index.php/apps/collectives/Link%20Preview%20Testing/Link%20Target?fileId=${pageId}`,
					`${baseUrl}/index.php/apps/collectives/Link%20Preview%20Testing/Link%20Target`,
					`${baseUrl}/index.php/apps/collectives/p/qqqYoCgYRnZ598p/Link%20Preview%20Testing/Link%20Target?fileId=${pageId}`,
					`${baseUrl}/index.php/apps/collectives/p/qqqYoCgYRnZ598p/Link%20Preview%20Testing/Link%20Target`,
				])
				.then(pageUrls => {
```

Reply:
agreed, good catch!